evidence.rkt

```diff
@@ -14,21 +14,48 @@
 
 (provide
   valid-id?
-  id->url
+  id->type/url
   add-last-publication-date
   expand-evidence
   make-pmid-expander
   make-nct-expander)
 
 (define (valid-id? id)
-  (regexp-match #rx"^PMID:[0-9]+$" id))
+  (or (string-prefix? id "PMID")
+      (string-prefix? id "NCT")
+      (string-prefix? id "clinicaltrialsNCT")
+      (string-prefix? id "DOI")))
 
-(define (id->url id)
-  (define split-id (string-split id ":"))
-  (and (not (null? (cdr split-id)))
-       (case (car split-id)
-         (("PMID") (format "https://pubmed.ncbi.nlm.nih.gov/~a" (cadr split-id)))
-         (else "Unknown"))))
+(define (id->type/url id)
+  (define (strip-id-tag id)
+    (define stripped-id (string-split id ":"))
```

Comment:
PSA: `string-split` may not behave the way you'd expect when one side of the separator is empty.  But you can provide the keyword argument `#:trim? #f` to normalize its behavior in this case.

For example:
```
> (string-split "x:" ":")
("x")
> (string-split "x:" ":" #:trim? #f)
("x" "")
```

We've been bitten by this before, so make sure you are getting the behavior you expect.